tests/build/build.go

```diff
@@ -516,6 +517,250 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build", "
+
+					if buildStatus.PaC != nil {
+						GinkgoWriter.Printf("state: %s\n", buildStatus.PaC.State)
+						GinkgoWriter.Printf("mergeUrl: %s\n", buildStatus.PaC.MergeUrl)
+						GinkgoWriter.Printf("errId: %d\n", buildStatus.PaC.ErrId)
+						GinkgoWriter.Printf("errMessage: %d\n", buildStatus.PaC.ErrMessage)
+						GinkgoWriter.Printf("configurationTime: %d\n", buildStatus.PaC.ConfigurationTime)
+					} else {
+						GinkgoWriter.Println("build status does not have PaC field")
+					}
+
+					return buildStatus.PaC != nil && buildStatus.PaC.State == "enabled" && buildStatus.PaC.MergeUrl != "" && buildStatus.PaC.ErrId == 0 && buildStatus.PaC.ConfigurationTime != "", nil
+				}, timeout, interval).Should(BeTrue(), "build status has unexpected content")
+			})
+
+			It("ensures build pipeline is triggered after merge", func() {
+				var pipelinerunPullSha string
+				var mergeResult *github.PullRequestMergeResult
+				var prNumber int
+
+				By("finding init PaC PR")
+				Eventually(func() (bool, error) {
+					prs, err := f.AsKubeAdmin.CommonController.Github.ListPullRequests(helloWorldComponentGitSourceRepoName)
+					if err != nil {
+						GinkgoWriter.Printf("cannot list pull requests: %v\n", err)
+						return false, err
+					}
+
+					for _, pr := range prs {
+						if pr.Head.GetRef() == branchName {
+							prNumber = pr.GetNumber()
+							pipelinerunPullSha = pr.Head.GetSHA()
+							return true, nil
+						}
+					}
+					return false, nil
+				}, timeout, interval).Should(BeTrue(), "timed out when waiting for init PaC PR to be created")
+
+				By("merging init PaC PR")
+				mergeResult, err := f.AsKubeAdmin.CommonController.Github.MergePullRequest(helloWorldComponentGitSourceRepoName, prNumber)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				By("checking both PipelineRuns exist and finish")
+				Expect(f.AsKubeAdmin.HasController.WaitForComponentPipelineToBeFinished(component, pipelinerunPullSha, 2)).To(Succeed(), "PipelineRun on pull request did not finish successfully")
+				Expect(f.AsKubeAdmin.HasController.WaitForComponentPipelineToBeFinished(component, mergeResult.GetSHA(), 2)).To(Succeed(), "PipelineRun on push did not finish successfully")
+
+				Expect(f.AsKubeAdmin.TektonController.DeleteAllPipelineRunsInASpecificNamespace(testNamespace)).To(Succeed())
+			})
+
+			It("can trigger simple build manually in PaC mode", func() {
+				Expect(f.AsKubeAdmin.HasController.SetComponentAnnotation(componentName, controllers.BuildRequestAnnotationName, controllers.BuildRequestTriggerSimpleBuildAnnotationValue, testNamespace)).To(Succeed())
+
+				Expect(f.AsKubeAdmin.HasController.WaitForComponentPipelineToBeFinished(component, "", 2)).To(Succeed())
+
+				Expect(f.AsKubeAdmin.TektonController.DeleteAllPipelineRunsInASpecificNamespace(testNamespace)).To(Succeed())
+			})
+
+			// Invalid request is currently treated as simple build trigger
+			It("handles invalid request annotation", func() {
+				Skip("until STONEBLD-1315 is merged")
```

Comment:
Did get it why we need to skip it now? STONEBLD-1315 is about removing the old annotations